src/weapon/w_032.c

```diff
@@ -9,7 +9,75 @@ extern s32 g_HandId;
 #define g_Animset w_032_1
 #define g_Animset2 w_032_2
 
-INCLUDE_ASM("weapon/nonmatchings/w_032", EntityWeaponAttack);
+extern WeaponAnimation D_E4000_8017A8F8[];
+
+void EntityWeaponAttack(Entity* self) {
+    WeaponAnimation* anim;
+    s16 animIndex;
+
+    self->posX.val = PLAYER.posX.val;
+    self->posY.val = PLAYER.posY.val;
+    self->facingLeft = PLAYER.facingLeft;
+    animIndex = (self->params >> 8) & 0x7F;
+    anim = &D_E4000_8017A8F8[animIndex];
+
+    if ((PLAYER.ext.player.anim < anim->frameStart ||
+         PLAYER.ext.player.anim >= (anim->frameStart + 7) ||
+         g_Player.unk46 == 0) &&
+        self->step != 2) {
+        self->step = 2;
+        self->ext.timer.t = 4;
```

Comment:
Are we using ext.timer rather than ext.weapon for usages like this? ext.weapon has pointers so it's not guaranteed to align as far as I know https://github.com/Xeeynamo/sotn-decomp/blob/ba13846faae69260e3f3603f119dbe1b02060357/include/entity.h#L200